indigo_app/forms.py

```diff
@@ -897,3 +932,30 @@ class FindPubDocForm(forms.Form):
     trusted_url = forms.URLField(required=False, widget=forms.HiddenInput())
     size = forms.IntegerField(required=False, widget=forms.HiddenInput())
     mimetype = forms.CharField(required=False, widget=forms.HiddenInput())
+
+
+class RepealMadeForm(forms.Form):
+    repealed_work = forms.ModelChoiceField(queryset=Work.objects)
+    repealed_date = forms.DateField(required=False, widget=forms.DateInput(attrs={'type': 'date'}))
+
+    def __init__(self, work, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.work = work
+
+    @cached_property
+    def repealed_work_obj(self):
+        return Work.objects.filter(pk=self['repealed_work'].value()).first()
+
+    def is_repealed_work_saved(self):
+        return self.repealed_work_obj.repealed_by == self.work
+
+    def save(self):
+        work = self.cleaned_data['repealed_work']
+        if self.cleaned_data.get('DELETE'):
+            work.repealed_by = None
+            work.repealed_date = None
+            work.save()
```

Comment:
I feel like this (and the save below) should be save_with_revision?

Reply:
agreed - @actlikewill pls add in your next pr